backend/uclapi/common/cachet.py

```diff
@@ -0,0 +1,151 @@
+import cachetclient.cachet as cachet
+import json
+from django.conf import settings
+
+"""
+This file handles all things cachet. From creating incidents to deleting
+them when fixed. The results can be seen on our cachet status dashboard at
+https://cachet.apps.uclapi.com/ .
+
+Incidents can be added to components of which we currently have 6 indicating
+problems with that specific system.
+List of components:
+Gencache-Staging,
+Occupeye-Staging,
+Occupeye-Mini-Staging,
+Gencache-Prod,
+Occupeye-Prod,
+Occupeye-Mini-Prod
+
+You can add a component in the admin dashboard and then reference it using
+the functions in this file without finding out its ID or attached information.
+
+More documentation on cachet endpoints can be found at
+https://docs.cachethq.io/reference
+and general documentation at
+https://docs.cachethq.io/docs
+"""
+
+
+class CachetException(Exception):
+    pass
+
+
+def get_incident_name(base):
+    """
+    Get and append the appropriate suffix for the component using the current
+    domain. For example if we are running on staging.ninja we add the
+    '-staging' suffix.
+
+    :param base: Base incident name
+    :type base: str
```

Comment:
Yeah I think in this case type hints would be more future proof for `:type var:` and `:rtype:`